### PR TITLE
Sdk upgrade changes for appframework

### PR DESCRIPTION
**Pull request by vivekr-splunk**
app framework code merged with develop branch. now operator sdk 1.15 is made to work for appframework.

### CODE REVIEW

**Review by gaurav-splunk — APPROVED** [review]

File: api/v3/common_types.go [hunk 1]
```diff
@@ -122,10 +127,12 @@ type CommonSplunkSpec struct {
 
 	// ReadinessInitialDelaySeconds defines initialDelaySeconds(See https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-readiness-probes) for Readiness probe
 	// Note: If needed, Operator overrides with a higher value
+	// +kubebuilder:validation:Minimum=0
```
Comment (sgontla) [inline api/v3/common_types.go:130]: is it optional by default? OR do we need an explicit market for that?

File: api/v3/common_types.go [hunk 2]
```diff
@@ -138,6 +145,8 @@ type StorageClassSpec struct {
 	StorageCapacity string `json:"storageCapacity"`
 
 	// If true, ephemeral (emptyDir) storage will be used
+	// default false
+	// +optional
```
Comment (sgontla) [inline api/v3/common_types.go:149]: it's new for me. does this notation work for the marker? similarly to other places

File: config/rbac/role.yaml [hunk 3]
```diff
@@ -0,0 +1,303 @@
+
```
Comment (sgontla) [inline config/rbac/role.yaml:1]: nit: remove empty space

File: api/v3/common_types.go (hunk 2 above)
Comment (vivekr-splunk) [reply api/v3/common_types.go:149]: its been used in k8s controller code

File: api/v3/common_types.go (hunk 1 above)
Comment (vivekr-splunk) [reply api/v3/common_types.go:130]: its mandatory field, if its set less than 0 than it should throw 400 error when tried to create CR. CR creation should fail

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    USER_NAME=nonroot
+
+RUN yum -y install shadow-utils
+RUN useradd -ms /bin/bash nonroot -u 1001
+
+LABEL name="splunk" \
+      maintainer="support@splunk.com" \
+      vendor="splunk" \
+      version="1.1.0" \
+      release="1" \
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply Dockerfile:38]: its just label, i am just using whatever the lables we had in 1.0.5

File: .github/workflows/pre-release-workflow.yml [hunk 5]
```diff
@@ -27,7 +27,11 @@ jobs:
   automated-release:
     name: Automated Pre Release
     runs-on: ubuntu-latest
-    if: github.ref == 'refs/heads/develop'
+    if:
+      contains('
+        refs/heads/develop
+        refs/heads/sdk-release-changes
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply .github/workflows/pre-release-workflow.yml:33]: removed, it was from some other branch I created

File: Makefile [hunk 6]
```diff
@@ -72,38 +272,37 @@ setup_clair_scanner: stop_clair_scanner
...
-lint:
-	@golint ./...
+#############################
+export ARCH=$(case $(uname -m) in x86_64) echo -n amd64 ;; aarch64) echo -n arm64 ;; *) echo -n $(uname -m) ;; esac)
+export OS=$(uname | awk '{print tolower($0)}')
+GO_DOWNLOAD_URL=https://go.dev/dl/go1.17.7.darwin-amd64.pkg
+export OPERATOR_SDK_DL_URL=https://github.com/operator-framework/operator-sdk/releases/download/v1.17.0
+OPERATOR_SDK_DOWNLOAD_URL=curl -LO ${OPERATOR_SDK_DL_URL}/operator-sdk_${OS}_${ARCH}
+MINIKUBE_DOWNLOAD_URL=https://storage.googleapis.com/minikube/releases/latest/minikube-darwin-amd64
+KUBECTL_DOWNLOAD_URL="https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/darwin/amd64/kubectl"
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply Makefile:291]: fixed , this variables for setting up development system.

File: config/rbac/role.yaml (hunk 3 above)
Comment (vivekr-splunk) [reply config/rbac/role.yaml:1]: removed

File: .github/workflows/nightly-int-test-workflow.yml [hunk 7]
```diff
@@ -74,8 +63,8 @@ jobs:
       - name: Install Go Lint
         run: |
           go version
-          go get -u golang.org/x/lint/golint
-      - name: Install Ginkgo
+          go install golang.org/x/lint/golint
+      - name: Install Ginkgo
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply .github/workflows/nightly-int-test-workflow.yml:67]: fixed

File: .github/workflows/int-test-workflow.yml [hunk 8]
```diff
@@ -125,17 +126,23 @@ jobs:
...
+      - name: install k8s dashboard
+        run: |
+          kubectl apply -f https://raw.githubusercontent.com/kubernetes/dashboard/v2.0.5/aio/deploy/recommended.yaml
       - name: Run Integration test
         run: |
           make int-test
       - name: Collect Test Logs
         if: ${{ always() }}
-        run: |
+        run: |
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply .github/workflows/int-test-workflow.yml:140]: fixed

File: .github/workflows/int-test-workflow.yml [hunk 9]
```diff
@@ -25,7 +25,7 @@ jobs:
     runs-on: ubuntu-latest
     env:
       CLUSTER_NODES: 1
-      CLUSTER_WORKERS: 3
+      CLUSTER_WORKERS: 5
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply .github/workflows/int-test-workflow.yml:28]: used for testing some performance testing, moved back to 3 nodes

File: .github/workflows/int-test-workflow.yml [hunk 10]
```diff
@@ -4,7 +4,7 @@ on:
     branches:
       - develop
       - master
-      - feature-appframework
+      - sdk-upgrade-feature-appframework
```
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply .github/workflows/int-test-workflow.yml:7]: yes I think so, i am adding sdk-upgrade feature branch to integration test pipeline

File: Makefile (hunk 6 above)
[the remark this answers is not in the document]
Comment (vivekr-splunk) [reply Makefile]: will testing

**Review by sgontla — APPROVED** [review]

**Review by smohan-splunk — APPROVED** [review]

**Pull request merged**